src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -343,7 +338,33 @@ public static void writeBundle(OutputStream outputStream, final List<Component>
         }
     }
 
-    private static Optional<ContentFilter> getContentFilter() throws IOException {
+    /**
+     * Filter the name of a content depending on the tokens in the name that need to be replaced.
+     * @param maybeFilter an Optional with a {@link ContentFilter} or not
+     * @param name the name of the content to be filtered
+     * @param tokens tokens in the name to be filtered. If null, the whole name is filtered
+     * @return the name filtered
+     */
+    private static String getNameFiltered(Optional<ContentFilter> maybeFilter, String name, String[] tokens) {
+        String filteredName;
+
+        if (tokens != null) {
+            // Filter each token or return the token depending on whether the filter is active or not
+            String[] replacedTokens = Arrays.stream(tokens).map(token -> maybeFilter.map(filter -> filter.filter(token)).orElse(token)).toArray(String[]::new);
+
+            // Replace each placeholder {0}, {1} in the name, with the replaced token
+            filteredName = MessageFormat.format(name, replacedTokens);
+        } else {
+            // Previous behavior was filter all the name, but it could end up in having a corrupted bundle. So we expect
+            // implementors to use the appropriate constructor of Content.
+            //filteredName = maybeFilter.map(filter -> filter.filter(name)).orElse(name);
+            filteredName = name;
+        }
+        System.out.format("%s is replaced by %s%n", name, filteredName);
```

Comment:
Not intented to be kept in production code right? :)

Reply:
No, sure